HxDropdownItem - add Icon parameter, default and documentation

vaclavek opened the pull request:
I noticed that the property 'Icon' is missing on HxDropdownItem. I added the property as it is implemented in HxContextMenuItem, including the CSS variable and documentation. Is there anything else I should do? :)

hakenr commented on Havit.Blazor.Components.Web.Bootstrap/wwwroot/defaults.css:
```diff
@@ -79,6 +80,9 @@
 	--hx-context-menu-item-icon-margin: 							0 .5rem 0 0;
 
+	/* HxDropdown */
+	--hx-dropdown-menu-item-icon: 									0 .5rem 0 0;
```
This should be named `--hx-dropdown-menu-item-icon-margin` (see the `HxContextMenuItem` variable few lines above).

hakenr reviewed (changes requested):
The `HxDropdownItem` is intended to be a "bare" Bootstrap component, and icons can be readily incorporated via the `ChildContent`. This is in contrast to `HxContextMenuItem`, which is a more comprehensive component that we define. Conversely, the `HxButton` does feature an `Icon` property. We could offer these enhancements to simplify the use of our component library.

However, I'm curious as to why you opted for `HxDropdownItem` to add the `Icon` property. Should this functionality also extend to `HxDropdownItemNavLink` and `HxDropdownItemText`? Perhaps even to `HxDropdownHeader`?

cc @crdo

crdo commented:
@vaclavek yes please. At least for HxDropdownItemText and HxDropdownItemNavLink.

fejfailo commented:
Hi @crdo and @hakenr,

I have added the Icon to the HxDropdownItemText, HxDropdownItemNavLink, and HxDropdownHeader components as requested. To be able to reuse the icon in the dropdown items component, I have added an internal component, HxDropdownIcon.

From my side, it is ready for review.

vaclavek commented:
Hi @crdo and @hakenr,

I have added the Icon to the HxDropdownItemText, HxDropdownItemNavLink, and HxDropdownHeader components as requested. To be able to reuse the icon in the dropdown items component, I have added an internal component, HxDropdownIcon.

From my side, it is ready for review

hakenr commented on Havit.Blazor.Components.Web.Bootstrap/Dropdowns/Internal/HxDropdownIcon.razor.cs:
```diff
@@ -0,0 +1,14 @@
+﻿namespace Havit.Blazor.Components.Web.Bootstrap.Internal;
+
+public partial class HxDropdownIcon : ComponentBase
```
All internal components share the `Internal` suffix in name.

hakenr commented on Havit.Blazor.Components.Web.Bootstrap/Dropdowns/Internal/HxDropdownIcon.razor.cs:
```diff
@@ -0,0 +1,14 @@
+	/// <summary>
+	/// Additional CSS class(es) for the context menu item icon.
+	/// </summary>
+	[Parameter] public string IconCssClass { get; set; }
```
Renaming to `CssClass`.

hakenr commented on Havit.Blazor.Documentation/Pages/Components/HxDropdownDoc/HxDropdown_Demo_WithIcons.razor:
```diff
@@ -0,0 +1,8 @@
+﻿<HxDropdownButtonGroup>
+    <HxDropdownToggleButton Color="ThemeColor.Secondary">Dropdown button</HxDropdownToggleButton>
+    <HxDropdownMenu CssClass="dropdown-menu-dark">
```
I will remove the "dark"-feature from this demo.

hakenr reviewed (approved):
lgtm (did some minor adjustments and merging)
